### DISCUSSION TITLE
Confirmation email

### DISCUSSION THREAD

**Discussion by ricardosikic** [supabase/supabase#22157] [Questions]
I have an app that has two types of user registrations: one for organizations and another for regular users. The problem arises because the confirmation email from Supabase only takes one URL, which is for confirming the regular user, but the organization user should have another URL for confirmation. Is it possible to make that confirmation email URL dynamic?

**Comment by GaryAustin1** [comment]
Sort of a hacky way might be possible.  https://github.com/orgs/supabase/discussions/21227
It makes use of user metadata in the signup which gets passed to the template.  The user metadata is not secure though, so may not work for your case.

**Reply by ricardosikic** [reply]
thanks for the documentation and answer @GaryAustin1 but it doesnt work. Using the confirmation url like this `<a href="{{ .Data.ConfirmationURL }}my-account?accepted=true"` goes to the url given but does not confirm the email unless i  keep the default config `<a href="{{ .ConfirmationURL }}my-account?accepted=true"...`

**Reply by ricardosikic** [reply]
Hi @GaryAustin1 i ended up using the function below as you told me a few days ago, thanks!
```
const { data, error } = await supabase.auth.admin.createUser({
      email: email,
      password: 'xxxxx',
      email_confirm: true,
      user_metadata: { username: name }
    })
```
